sdk/typespec/typespec_client_core/src/fmt.rs

```diff
@@ -25,7 +30,7 @@ pub use typespec_macros::SafeDebug;
 /// };
 /// assert_eq!(format!("{model:?}"), "MyModel { .. }");
 /// ```
-pub trait SafeDebug: Debug {}
+pub trait SafeDebug: crate::__private::Sealed {}
```

Comment:
Doesn't sealing the trait make it impossible for other crates to impl, even via `#[derive]`?

Reply:
Ah, I see (and/or remember from earlier). The trait isn't really intended to be derived, it's just a marker trait for the derive macro to run.

It might be confusing to have the trait at all though, since using it in a bound would not have the desired effect of limiting a generic type to ones that have implemented `SafeDebug`. It also makes it more difficult to provide any methods on `SafeDebug` to "unredact" the debug output later.

Reply:
I tried to remove it but seems a derive macro doesn't like that. But, after a quick search, it seems it's possible. Let me try. I definitely wanted to remove the trait to avoid confusion. That's why I ended up sealing it instead.